### PR TITLE
Catalog items v20220401

### PR DESCRIPTION
**Pull request by prashant-thorat17**
Provided Catalog items api's of version 2022-04-01.

### CODE REVIEW

File: lib/muffin_man/catalog_items/v20220401.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module MuffinMan
+  module CatalogItems
+    class V20220401 < V20201201
```
Comment (jasonwells) [inline lib/muffin_man/catalog_items/v20220401.rb:3]: If you're going to inherit here, I think it'd be better to have a base class, and then have both versions inherit from the base class.

File: lib/muffin_man/catalog_items/v20201201.rb [hunk 2]
```diff
@@ -17,7 +17,7 @@ class V20201201 < SpApiClient
       ].freeze
       GET_CATALOG_ITEM_PARAMS = %w[includedData locale].freeze
 
-      def search_catalog_items(keywords, marketplace_ids, params = {})
+      def search_catalog_items(keywords, marketplace_ids, params = {}, version="2020-12-01")
```
Comment (jasonwells) [inline lib/muffin_man/catalog_items/v20201201.rb:20]: Again, don't make the version passed in to the already versioned class name. We want to keep the gems naming and structure consistent. If a base class is needed, implement that.

**Review by jasonwells — CHANGES_REQUESTED** [review]
A think just an inheritance problem here, if needed, let's move to a base class.

File: lib/muffin_man/catalog_items/v20220401.rb (hunk 1 above)
Comment (prashant-thorat17) [reply lib/muffin_man/catalog_items/v20220401.rb]: I have added base_api class for these API's

**Review by jasonwells — APPROVED** [review]

**Pull request merged**